src/transformers/training_args.py

```diff
@@ -1938,7 +1946,16 @@ def _setup_devices(self) -> "torch.device":
                     f"Using the `Trainer` with `PyTorch` requires `accelerate>={ACCELERATE_MIN_VERSION}`: "
                     "Please run `pip install transformers[torch]` or `pip install accelerate -U`"
                 )
-            AcceleratorState._reset_state(reset_partial_state=True)
+            use_configured_state = self.accelerator_config.pop("use_configured_state", False)
+            if use_configured_state:
+                if PartialState._shared_state == {}:
+                    raise ValueError(
+                        "Passing `'use_configured_state':True` to the AcceleratorConfig requires a pre-configured "
+                        "`AcceleratorState` or `PartialState` to be defined before calling `TrainingArguments`. "
+                        "Please define this beforehand."
```

Comment:
ultra-nit - we don't really need this line, as required behaviour is explain in the previous section. I don't mind if you want to leave it in

```suggestion
```